lib/parser.js

```diff
@@ -118,7 +118,7 @@ class Parser {
       })
       req.on('error', reject);
       timeout = setTimeout(() => {
-        return reject(new Error("Request timed out after " + this.options.timeout + "ms"));
+        req.destroy(new Error("Request timed out after " + this.options.timeout + "ms"));
```

Comment:
```suggestion
        let err = new Error("Request timed out after " + this.options.timeout + "ms");
        req.destroy(err);
        reject(err);
```

Reply:
req.destroy will also call the callback for req.on('error',...) above so no need to reject it sepratly here.